src/wp-admin/includes/class-wp-media-list-table.php

```diff
@@ -748,129 +748,87 @@ protected function get_default_primary_column_name() {
-						_x( 'Trash', 'verb' )
-					);
-				}
+		if ( $this->is_trash && current_user_can( 'delete_post', $post->ID ) || ! EMPTY_TRASH_DAYS || ! MEDIA_TRASH ) {
+			$actions['delete'] = sprintf(
+				'<a href="%s" class="submitdelete aria-button-if-js"%s aria-label="%s">%s</a>',
+				wp_nonce_url( "post.php?action=delete&amp;post=$post->ID", 'delete-post_' . $post->ID ),
+				! MEDIA_TRASH ? " onclick='return showNotice.warn();'" : '',
+				/* translators: %s: Attachment title. */
+				esc_attr( sprintf( __( 'Delete &#8220;%s&#8221; permanently' ), $att_title ) ),
+				__( 'Delete Permanently' )
+			);
+		}
 
-				if ( $this->is_trash || ! EMPTY_TRASH_DAYS || ! MEDIA_TRASH ) {
-					$delete_ays        = ( ! $this->is_trash && ! MEDIA_TRASH ) ? " onclick='return showNotice.warn();'" : '';
-					$actions['delete'] = sprintf(
-						'<a href="%s" class="submitdelete aria-button-if-js"%s aria-label="%s">%s</a>',
-						wp_nonce_url( "post.php?action=delete&amp;post=$post->ID", 'delete-post_' . $post->ID ),
-						$delete_ays,
-						/* translators: %s: Attachment title. */
-						esc_attr( sprintf( __( 'Delete &#8220;%s&#8221; permanently' ), $att_title ) ),
-						__( 'Delete Permanently' )
-					);
-				}
-			}
+		if ( ! $this->is_trash && get_permalink( $post->ID ) ) {
+			$actions['view'] = sprintf(
+				'<a href="%s" aria-label="%s" rel="bookmark">%s</a>',
+				get_permalink( $post->ID ),
+				/* translators: %s: Attachment title. */
+				esc_attr( sprintf( __( 'View &#8220;%s&#8221;' ), $att_title ) ),
+				__( 'View' )
+			);
+		}
 
-			if ( ! $this->is_trash ) {
-				if ( get_permalink( $post->ID ) ) {
-					$actions['view'] = sprintf(
-						'<a href="%s" aria-label="%s" rel="bookmark">%s</a>',
-						get_permalink( $post->ID ),
-						/* translators: %s: Attachment title. */
-						esc_attr( sprintf( __( 'View &#8220;%s&#8221;' ), $att_title ) ),
-						__( 'View' )
-					);
-				}
+		if ( $this->detached && current_user_can( 'edit_post', $post->ID ) ) {
+			$actions['attach'] = sprintf(
+				'<a href="#the-list" onclick="findPosts.open( \'media[]\', \'%s\' ); return false;" class="hide-if-no-js aria-button-if-js" aria-label="%s">%s</a>',
+				$post->ID,
+				/* translators: %s: Attachment title. */
+				esc_attr( sprintf( __( 'Attach &#8220;%s&#8221; to existing content' ), $att_title ) ),
+				__( 'Attach' )
+			);
+		}
 
-				$actions['copy'] = sprintf(
-					'<span class="copy-to-clipboard-container"><button type="button" class="button-link copy-attachment-url media-library" data-clipboard-text="%s" aria-label="%s">%s</button><span class="success hidden" aria-hidden="true">%s</span></span>',
-					esc_url( wp_get_attachment_url( $post->ID ) ),
-					/* translators: %s: Attachment title. */
-					esc_attr( sprintf( __( 'Copy &#8220;%s&#8221; URL to clipboard' ), $att_title ) ),
-					__( 'Copy URL' ),
-					__( 'Copied!' )
-				);
+		if ( ! $this->is_trash && wp_get_attachment_url( $post->ID ) ) {
```

Comment:
I know the aim is to keep each action in its own separate condition block, but I'm thinking we can reduce the calls to `wp_get_attachment_url()` here.

```php
$attachment_url = wp_get_attachment_url( $post->ID );
if ( $attachment_url ) {
	if ( ! $this->is_trash ) {
		$actions['copy'] = sprintf(
			'<span class="copy-to-clipboard-container"><button type="button" class="button-link copy-attachment-url media-library" data-clipboard-text="%s" aria-label="%s">%s</button><span class="success hidden" aria-hidden="true">%s</span></span>',
			esc_url( $attachment_url ),
			/* translators: %s: Attachment title. */
			esc_attr( sprintf( __( 'Copy &#8220;%s&#8221; URL to clipboard' ), $att_title ) ),
			__( 'Copy URL' ),
			__( 'Copied!' )
		);
	}

	$actions['download'] = sprintf(
		'<a href="%s" aria-label="%s" download>%s</a>',
		esc_url( $attachment_url ),
		/* translators: %s: Attachment title. */
		esc_attr( sprintf( __( 'Download &#8220;%s&#8221;' ), $att_title ) ),
		__( 'Download file' )
	);
}
```
What do you think @kebbet?

Reply:
While it's reducing calls to `wp_get_attachment_url()` it nests two action in one conditional check. I don't like the look of it, but my opinion on that is just opinonated. Reduced number of calls is better I guess.